integration_test/tests/disconnect_tests.rs

```diff
@@ -0,0 +1,130 @@
+    use definitions::{
+        AttrOdbcVersion, FreeStmtOption, Handle, HandleType, SQLExecDirectW, SQLFreeHandle,
+        SQLFreeStmt, SqlReturn, SQL_NTS,
+    };
+
+    #[test]
+    fn sql_disconnect_closes_statement_implicitly() {
+        let (env, dbc, stmt) = default_setup_connect_and_alloc_stmt(AttrOdbcVersion::SQL_OV_ODBC3);
+        let query = b"SELECT * FROM integration_test.foo\0".map(|b| b as u16);
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLExecDirectW(stmt, query.as_ptr(), SQL_NTS as i32,),
+                "{}",
+                get_sql_diagnostics(HandleType::SQL_HANDLE_STMT, *stmt.cast::<Handle>()),
+            );
+
+            // we will not free the statement, instead proceeding directly to disconnect and freeing the dbc and env handles
+            disconnect_and_free_dbc_and_env_handles(env, dbc)
+        }
+    }
+
+    #[test]
+    fn sql_disconnect_implicitly_closing_statment_cursors_is_safe_for_closed_statements() {
+        let (env, dbc, stmt) = default_setup_connect_and_alloc_stmt(AttrOdbcVersion::SQL_OV_ODBC3);
+        let query = b"SELECT * FROM integration_test.foo\0".map(|b| b as u16);
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLExecDirectW(stmt, query.as_ptr(), SQL_NTS as i32,),
+                "{}",
+                get_sql_diagnostics(HandleType::SQL_HANDLE_STMT, *stmt.cast::<Handle>()),
+            );
+
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLFreeStmt(stmt, FreeStmtOption::SQL_CLOSE as i16),
+                "{}",
+                get_sql_diagnostics(HandleType::SQL_HANDLE_STMT, *stmt.cast::<Handle>()),
+            );
+
+            // we will not free the statement, instead proceeding directly to disconnect and freeing the dbc and env handles
+            disconnect_and_free_dbc_and_env_handles(env, dbc)
+        }
+    }
+
+    #[test]
+    fn sql_disconnect_implicitly_closing_statment_cursors_is_safe_for_unbound_statemens() {
+        let (env, dbc, stmt) = default_setup_connect_and_alloc_stmt(AttrOdbcVersion::SQL_OV_ODBC3);
+        let query = b"SELECT * FROM integration_test.foo\0".map(|b| b as u16);
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLExecDirectW(stmt, query.as_ptr(), SQL_NTS as i32,),
+                "{}",
+                get_sql_diagnostics(HandleType::SQL_HANDLE_STMT, *stmt.cast::<Handle>()),
+            );
+
+            // we didn't actually bind a statement, so this should be a no-op
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLFreeStmt(stmt, FreeStmtOption::SQL_UNBIND as i16),
+                "{}",
+                get_sql_diagnostics(HandleType::SQL_HANDLE_STMT, *stmt.cast::<Handle>()),
+            );
+
+            // we will not free the statement, instead proceeding directly to disconnect and freeing the dbc and env handles
```

Comment:
```suggestion
            // Directly disconnect and free the dbc and env handles without closing the statement
```
We should either get rid of this comment or change the word "free" to "close" since we do call `SQLFreeStmt` but use it to unbind rather than close.